make: Updated Makefile.am to support distcheck

sshambar opened the pull request:
Updated Makefile.am to allow the make distcheck to work, which
means make install can be reversed by make uninstall, and
make clean correctly removes generated files.  make dist now
also creates a tarfile that can be used to configure and
build from.  Added README to install.

Signed-off-by: Scott Shambarger <devel@shambarger.net>

commodo commented:
is this adding new build rules to `make`
if yes, can we add some CI calls to this in the [build.yml file](https://github.com/mstpd/mstpd/blob/master/.github/workflows/build.yml)

i can add them, if you tell me the make commands

sshambar commented:
Well, it's not adding a target so much as fixing one :)  I added a CI entry for "make distcheck" and it passed on my fork.

I also added LICENSE to dist, and removed a couple generated files that shouldn't have been in the tar.

PaulSD commented on Makefile.am:
```diff
@@ -102,7 +128,7 @@ rpm: rpm.spec
 	tar -cvjf "mstpd-$$Version.tar.bz2" "mstpd-$$Version" ; \
 	rm -rf "mstpd-$$Version" ; \
-	cd .. ; mkdir -p SPECS ; cp "$$SrcDir/rpm.spec" SPECS/mstpd.spec
+	cd .. ; mkdir -p SPECS ; cp "$$SrcDir/rpm.spec" SPECS/mstpd.spec; \
```
This will cause a merge conflict with #119 ... You might want to skip it here to avoid that.

sshambar replied:
Cool... was just fixing bug in the file (and perhaps 119 wouldn't be merged :), reverted the fix.

PaulSD reviewed (approved):
LGTM
